### DISCUSSION TITLE
umi3升级到umi4之后，antd升级到5，做了样式兼容，但是使用不同的安装命令打包之后效果不一样

### DISCUSSION THREAD

**Discussion by jiajiaxing1** [umijs/umi#12348] [Q&A]
参考https://ant-design.antgroup.com/docs/react/compatible-style-cn
这个文档做的样式兼容
使用cnpm安装依赖打包后 在Chrome79上样式异常
使用pnpm安装依赖打包后 在Chrome79上样式正常
查看了antd和cssinjs的安装版本是相同的  不确定是不是两个安装方式的依赖关系影响的

**Comment by fz6m** [comment]
这个问题可能要考虑多种原因，比如 umi 的版本是不是最新的 、antd 的版本、是否开启了 antd 5 样式降级的选项、如何配置开启的、降级使用的 antd css in js 包的版本是不是和 antd 依赖的一样、每次安装依赖是不是删了 `node_modules` 再安装的、pnpm 的版本等等。

如果有最小复现的话，提供一个最小复现看看吧。

**Reply by jiajiaxing1** [reply]
umi版本是4.1.10
antd@5.8.6
cssinjs@1.16.0
antd和cssinjs都是锁版本了的
在config里配置了 开启styleProvider
antd: {
   
    configProvider: {},

    styleProvider: {
      hashPriority: 'high',
      legacyTransformer: true,
    },
  },

初步发现使用cnpm安装依赖后:where选择器还是存在
使用pnpm安装并启动:where选择器切换成类选择器了

**Reply by jiajiaxing1** [reply]
cnpm 是9.2.0
pnpm 是8.9.2

**Reply by jiajiaxing1** [reply]
解决了 之前是因为怕版本cssinjs和antd版本不一致所以锁版本了
然后看了antd里面的cssinjs是没有锁版本的 就使用 "@ant-design/cssinjs": "^1.16.0",重新用cnpm安装依赖就没问题了

**Reply by fz6m** [reply]
你用 `pnpm why @ant-design/cssinjs` 可以看到 antd 依赖的那个 cssinjs 的依赖版本号，然后安装一样的就可以了，或者用 pnpm overrides 去全局固定一下，这样不会因为 lock 文件被删除重新安装导致的不一致（通常来说 lock 文件不会被删除）。

**Reply by jiajiaxing1** [reply]
好的
